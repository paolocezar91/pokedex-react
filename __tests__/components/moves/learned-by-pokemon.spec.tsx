import LearnedByPokemon from '@/components/moves/learned-by-pokemon';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { INamedApiResource, IPokemon } from 'pokeapi-typescript';

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

jest.mock('@/context/UserContext', () => ({
  useUser: () => ({
    settings: { descriptionLang: 'en' }
  })
}));

describe('LearnedByPokemon Component', () => {
  const mockLearnedByPokemon = [
    {
      "name": "ditto",
      "url": "/api/v2/pokemon/132/"
    },
    {
      "name": "mew",
      "url": "/api/v2/pokemon/151/"
    }
  ] as unknown as INamedApiResource<IPokemon>[];

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render the component with initial state', () => {
    render(<LearnedByPokemon learnedByPokemon={mockLearnedByPokemon} />);

    // Test title transformation
    expect(screen.getByRole('heading')).toHaveTextContent('moves.learnedBy.title');
    expect(screen.getByText('Mew')).toBeInTheDocument();
    expect(screen.getByText('Ditto')).toBeInTheDocument();
  });

  it('should have the link to the pokemons', () => {
    render(<LearnedByPokemon learnedByPokemon={mockLearnedByPokemon} />);

    // Test title transformation
    expect(screen.getByRole('heading')).toHaveTextContent('moves.learnedBy.title');
    expect(screen.getByText('Mew')).toBeInTheDocument();
    const mewLink = screen.getByRole('link', { name: 'Mew' });
    expect(mewLink).toHaveAttribute('href', '/pokedex/mew');
  });


  it('should match snapshot', () => {
    const { asFragment } = render(<LearnedByPokemon learnedByPokemon={mockLearnedByPokemon} />);
    expect(asFragment()).toMatchSnapshot();
  });
});