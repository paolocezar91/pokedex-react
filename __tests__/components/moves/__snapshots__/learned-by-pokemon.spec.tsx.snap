// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnedByPokemon Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-fit learned-by-pokemon w-full flex flex-col flex-1 h-0 mt-2"
  >
    <h3
      class="w-fit text-lg mb-4"
    >
      moves.learnedBy.title
    </h3>
    <div
      class="sm:overflow-initial md:overflow-auto flex-1 pr-4"
    >
      <table
        class="w-full text-xs"
      >
        <thead>
          <tr
            class="sticky top-0 bg-background z-1"
          >
            <th
              class="w-[0%] text-white text-center px-2 py-2"
            >
              #
            </th>
            <th
              class="w-[5%] text-white text-left px-2 py-2"
            >
              table.name
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="border-solid border-foreground  border-b-2"
          >
            <td
              class="p-2"
            >
              #0132
            </td>
            <td
              class="p-2"
            >
              <a
                class="hover:bg-(--pokedex-red-dark) p-1"
                href="/pokedex/ditto"
              >
                Ditto
              </a>
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="p-2"
            >
              #0151
            </td>
            <td
              class="p-2"
            >
              <a
                class="hover:bg-(--pokedex-red-dark) p-1"
                href="/pokedex/mew"
              >
                Mew
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
